src/tests/ftest/cart/util/cart_logparse.py

```diff
@@ -535,7 +535,12 @@ def _load_data(self):
                 if l_pid in pids:
                     pids[l_pid]['line_count'] += 1
                 else:
-                    pids[l_pid] = {'line_count': 1,
+                    if l_pid != int(-1) and -1 in pids:
```

Comment:
Is this the bit that's new?  If so can you comment on what it's doing please?

If I remember this is the part that creates some metadata about pids and their first character and line count in the file so it can be parsed from disk, but there are two other implementations of this, one for small files where the entire file is loaded into memory, and one for bzip files where it's walked iteratively but you can't seek() so it has to start at the beginning each time.

Reply:
Inside the log, the rank(pid) is -1 before the node is assigned to the real rank. So this is for checking if the rank is assigned (!= -1) and there -1 in pid(previous log),  then let's merge them together.